src/client/keycode.h

```diff
@@ -4,62 +4,70 @@
 
 #pragma once
 
-#include "exceptions.h"
 #include "irrlichttypes.h"
+#include "cmake_config.h" // USE_SDL2
 #include <Keycodes.h>
 #include <IEventReceiver.h>
 #include <string>
+#include <unordered_map>
+#include <variant>
 
-class UnknownKeycode : public BaseException
-{
-public:
-	UnknownKeycode(const char *s) :
-		BaseException(s) {};
-};
-
-/* A key press, consisting of either an Irrlicht keycode
-   or an actual char */
-
+/* A key press, consisting of a scancode or a keycode */
 class KeyPress
 {
 public:
 	KeyPress() = default;
 
-	KeyPress(const char *name);
+	KeyPress(std::string_view name);
 
-	KeyPress(const irr::SEvent::SKeyInput &in, bool prefer_character = false);
+	KeyPress(const irr::SEvent::SKeyInput &in);
 
-	bool operator==(const KeyPress &o) const
+	std::string sym() const;
+	std::string name() const;
+
+	irr::EKEY_CODE getKeycode() const;
```

Comment:
can we document these with short comments? doesn't have to be the full doxygen form

Reply:
https://github.com/luanti-org/luanti/pull/14964/commits/b349b7aa21e55be312d9a1dc1e45947202b7b448